ttnn/cpp/ttnn/operations/pool/pool_utils.cpp

```diff
@@ -1,22 +1,104 @@
-// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
-//
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
 // SPDX-License-Identifier: Apache-2.0
 
 #include "pool_utils.hpp"
 #include <limits>
 #include <tt-metalium/bfloat16.hpp>
 #include <tt-metalium/assert.hpp>
+#include <algorithm>
 
 namespace ttnn::operations::pool {
+// This function generates a vector of elements of type ScalarInfo. It is called once per core and generates the
+// adequate scalars for each output element that core should produce. It should be called only for avg pool operation
+// and only if the divisor_override is NOT set and the idea behind it is to generate config tesnor in cases where one
+// scalar per core is not sufficient to create correct result. Those scenarios are ceil_mode == true and (ceil_pad_h > 0
+// || ceil_pad_w > 0) or count_include_pad == false || (pad_h > 0 || pad_w > 0). Both of these scenarios can be
+// irrelevant if the divisor_override is set, in which case we don't calculate the divisor since it is already passed as
+// an argument. It only adds scalars that are different than the scalar preeceding it not to have duplicates of data,
+// this is why we use start and end indices to know how many sequential output elements should be multiplied by the same
+// scalar value.
+std::vector<ScalarInfo> get_bf16_avg_pool_config_scalars(
```

Comment:
is there a need for this to live here in utility if it's used only in one place?